src/Microsoft.Framework.PackageManager/ConsoleCommands/SourceConsoleCommand.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.Framework.PackageManager.SourceControl;
+using Microsoft.Framework.Runtime.Common.CommandLine;
+
+namespace Microsoft.Framework.PackageManager
+{
+    internal class SourceConsoleCommand
+    {
+        public static void Register(CommandLineApplication cmdApp, ReportsFactory reportsFactory)
+        {
+            cmdApp.Command("source", c =>
+            {
+                c.Description = "Retrieves the source code for packages used by projects";
+
+                var packagesArgument = c.Argument(
+                    "[package]",
+                    "The name of the package for which to retrieve the sources. Can only specify packages used by the project.",
+                    multipleValues: true);
+
+                var projectFileOption = c.Option(
+                    "-p|--project",
+                    "Optional. The path to a project.json file. If not specified, the project in the current folder is used.",
+                    CommandOptionType.SingleValue);
+                var packagesFolderOption = c.Option(
+                    "-pa|--packages",
```

Comment:
Even if we support `-pa`, it is not a good practice. Choose another single-character name.